examples/threading/threading.c

```diff
@@ -14,6 +20,40 @@ void* threadfunc(void* thread_param)
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    int t1, t2, p1, p2;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    //sleep before acquiring mutex, for time specified in wait_to_obtain
+    t1 = usleep(thread_func_args->wait_to_obtain);
+    if(t1)
+    {
+    	ERROR_LOG("usleep1");
+    }
+    
+    //acquire the mutex
+    p1 = pthread_mutex_lock(thread_func_args->mutex);
+    if(p1)
+    {
+    	ERROR_LOG("mutex_lock");
+    }
+    
+    //release time of mutex specified in wait_to_obtain
+    t2 = usleep(thread_func_args->wait_to_release);
+    if(t2)
+    {
+    	ERROR_LOG("usleep2");
+    }
+    
+    //release the mutex
+    p2 = pthread_mutex_unlock(thread_func_args->mutex);
+    if(p2)
+    {
+    	ERROR_LOG("mutex_unlock");
+    }
+    
+    thread_func_args->thread_complete_success = true;
```

Comment:
@DivyeshShashikant in case of any of errors above, you are returning true here! do you think is this proper handling?